core/shared/src/main/scala/zio/internal/StackBool.scala

```diff
@@ -96,14 +96,20 @@ private[zio] final class StackBool private () {
 private[zio] object StackBool {
   def apply(): StackBool = new StackBool
 
-  def apply(bools: Boolean*): StackBool = {
+  def apply(bool: Boolean): StackBool = {
     val stack = StackBool()
 
-    bools.reverse.foreach(stack.push(_))
+    stack.push(bool)
 
     stack
   }
 
+  def apply(bools: List[Boolean]): StackBool = {
```

Comment:
This one should be called `fromIterable` and should take `Iterable[Boolean]` to match ZIO conventions.